### PR TITLE
Added problem 3: sumDigits to the problems page

### PR DESCRIPTION
**Pull request by lalaAdasia00**

### CODE REVIEW

File: problems/fibonacci.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 id: 2
 title: Fibonacci
 text: |
-  Write a function called Fibonacci that takes in an integer.
+  Write a function called Fibonacci that takes in an integer
```
Comment (nicholashusin) [inline problems/fibonacci.yaml]: I think you erased the `.` by accident?

File: problems/fizzbuzz.yaml [hunk 2]
```diff
@@ -32,4 +32,4 @@ solution: |
       return num;
 
     return str;
-  }
+  };
```
Comment (nicholashusin) [inline problems/fizzbuzz.yaml]: This semicolon is unnecessary and will actually trigger a warning from codemirror linter.

File: problems/sumDigits.yaml [hunk 3]
```diff
@@ -0,0 +1,32 @@
+id: 3
+title: SumDigits
+text: |
+  Write a function called sumDigits that takes in a non-negative ineteger.
+  given an input N, and returns that sum of all its digits. If the number.
```
Comment (nicholashusin) [inline problems/sumDigits.yaml]: I think you added a `.` by accident here.

File: problems/sumDigits.yaml [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  - assert(sumDigits(-123) === 0);
+  - assert(sumDigits(-456) === 0);
+solution: |
+    function sumDigits(num) {
+      let sum = 0;
+      while (n) {
+          digit = n % 10;
+          sum += digit;
+          n = (n - digit) / 10;
+        }
```
Comment (nicholashusin) [inline problems/sumDigits.yaml]: Inconsistent indentation: the `}` and the line below it is indented further than `while (n) {`.

**Review by nicholashusin — COMMENTED** [review]
Looks good overall.
There are some minor formatting issues that I think we should take care of though.

File: problems/sumDigits.yaml [hunk 5]
```diff
@@ -0,0 +1,32 @@
+id: 3
+title: SumDigits
+text: |
+  Write a function called sumDigits that takes in a non-negative ineteger.
```
Comment (ifeomi) [inline problems/sumDigits.yaml]:
```suggestion
  Write a function called sumDigits that takes in a non-negative integer.
```

File: problems/sumDigits.yaml [hunk 6]
```diff
@@ -0,0 +1,32 @@
+id: 3
+title: SumDigits
+text: |
+  Write a function called sumDigits that takes in a non-negative ineteger,
+  given an input num, and returns that sum of all its digits. If the number
+  is negative, the first digit should count as the only negative.
```
Comment (ifeomi) [inline problems/sumDigits.yaml]:
```suggestion
  Write a function called sumDigits that takes in an integer and returns the 
  sum of all its digits. If the number is negative, the first digit should count as 
  the only negative.
```
this reads more clearly to me :)

**Review by nicholashusin — APPROVED** [review]
LGTM, thanks :)

File: problems/sumDigits.yaml [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+  
+  Refer to example below.
+  
+  sumDigits(123456) = 1 + 2 + 3 + 4 + 5 + 6 = 21
+  sumDigits(-123456) = -1 + 2 + 3 + 4 + 5 + 6 = 19
+code: |
+  function sumDigits(num) {
+    return num;
+  }
```
Comment (ifeomi) [inline problems/sumDigits.yaml:15]: can you add test cases to the initial code like in PR #30?

File: problems/sumDigits.yaml [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+tests:
+  - assert(sumDigits(0) === 0);
+  - assert(sumDigits(123) === 6);
+  - assert(sumDigits(456) === 15);
+  - assert(sumDigits(789) === 24);
+  - assert(sumDigits(-123) === 0);
+  - assert(sumDigits(-456) === 0);
+solution: |
+    function sumDigits(num) {
+        let sum = 0;
```
Comment (ifeomi) [inline problems/sumDigits.yaml]: let's do 2 spaces for indentation [(per google's style guide)](https://google.github.io/styleguide/jsguide.html#formatting-block-indentation)

File: problems/sumDigits.yaml [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+        while (n) {
```
Comment (ifeomi) [inline problems/sumDigits.yaml]: i think this should be `num` not `n` throughout
```suggestion
        while (num) {
```

**Review by ifeomi — CHANGES_REQUESTED** [review]

File: problems/sumDigits.yaml (hunk 6 above)
Comment (lalaAdasia00) [reply problems/sumDigits.yaml]: Thanks. Making changes now

**Review by ifeomi — APPROVED** [review]
LGTM, ty!!

File: problems/sumDigits.yaml [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+  function sumDigits(num) {
+    return num;
+  }
+
+  assert(sumDigits(0) === 0);
+  assert(sumDigits(123) === 6);
+  assert(sumDigits(456) === 15);
+  assert(sumDigits(789) === 24);
+  assert(sumDigits(-123) === 0);
+  assert(sumDigits(-456) === 0);
```
Comment (chronologos) [inline problems/sumDigits.yaml]: is this right? 😀

File: problems/sumDigits.yaml [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+  alert('Passed all test cases!');
+tests:
+  - assert(sumDigits(0) === 0);
+  - assert(sumDigits(123) === 6);
+  - assert(sumDigits(456) === 15);
+  - assert(sumDigits(789) === 24);
+  - assert(sumDigits(-123) === 0);
+  - assert(sumDigits(-456) === 0);
+solution: |
+  function sumDigits(num) {
```
Comment (chronologos) [inline problems/sumDigits.yaml:32]: this seems to fail on negative numbers

**Review by chronologos — CHANGES_REQUESTED** [review]

File: problems/sumDigits.yaml (hunk 11 above)
Comment (lalaAdasia00) [reply problems/sumDigits.yaml:32]: I'm sorry. I forgot to change those .

Comment (chronologos) [reply problems/sumDigits.yaml:32]: i meant to say the code itself will not work on negative numbers.
sumDigits(-123) will give -6 instead of 4.

Comment (lalaAdasia00) [reply problems/sumDigits.yaml:32]: Ohhh I see !

**Review by chronologos — APPROVED** [review]

**Pull request merged**